src/System.Net.Security/src/System/Net/Security/SslState.cs

```diff
@@ -811,26 +823,31 @@ private void StartSendBlob(byte[] incoming, int count, AsyncProtocolRequest asyn
                 }
             }
 
-            CheckCompletionBeforeNextReceive(message, asyncRequest);
+            CheckCompletionBeforeNextReceive(message, asyncRequest, cancellationToken);
         }
 
         //
         // This will check and logically complete / fail the auth handshake.
         //
-        private void CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest)
+        private void CheckCompletionBeforeNextReceive(ProtocolToken message, AsyncProtocolRequest asyncRequest, CancellationToken cancellationToken)
         {
+            if (asyncRequest != null)
+            {
+                cancellationToken.ThrowIfCancellationRequested();
```

Comment:
What guarantees if any do we make if authentication is canceled in the middle of it?  Presumably no guarantees and at that point you can't use the SslStream for anything else?

Reply:
Yes, the exception thrown for cancellation will fail the handshake, and make the SslStream invalid.